packages/react-storage/src/components/StorageBrowser/context/actions/listLocationItemsAction.ts

```diff
@@ -40,13 +40,13 @@ export interface ListLocationItemsActionOutput {
 }
 
 const parseResultItems = (items: ListOutputItem[]): LocationItem[] =>
-  items.map(({ path: key, lastModified, size }) =>
-    lastModified
-      ? // `size` is marked as potentially `undefined` in `ListOutputItem`
-        // but is populated when the item is a file
-        { key, lastModified, size: size!, type: 'FILE' }
-      : { key, type: 'FOLDER' }
-  );
+  items.map(({ path: key, lastModified, size }) => {
+    if (size === 0 && key.endsWith('/')) {
```

Comment:
Added this just for now. In one of the items that gets returned, there is a `txt` file that is 0 bytes. So far it looks like the items that are folders do end with `/`, but is there something else we need to do to ensure this?

Reply:
Think we need something a bit more resilient but this is fine for now IMO